### PR TITLE
Add PR workflow for docs/pkg builds

### PR DESCRIPTION
**Pull request by pedro-psb**
Adds basic PR workflow for:

- package build
- docs build

This is based on pulp-cli workflow files. I've copied and commented some steps so I can work on them later.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+on:
+  workflow_call:
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+      # - uses: "actions/cache@v4"
```
Comment (dkliban) [inline .github/workflows/build.yml]: did you mean to leave this comment here?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+on:
+  workflow_call:
+
+jobs:
+  test:
+    runs-on: "ubuntu-20.04"
+    steps:
+      - uses: "actions/checkout@v4"
+      # - uses: "actions/cache@v4"
+      #   with:
```
Comment (dkliban) [inline .github/workflows/docs.yml]: same here?

Comment (pedro-psb) [reply .github/workflows/docs.yml]: Kind of. I've copied those from pulp-cli, but didn't want to bother setting up some things right now.

Comment (pedro-psb) [reply .github/workflows/docs.yml]: I could remove and just check the src again later.

Comment (dkliban) [reply .github/workflows/docs.yml]: i prefer to not leave commented out code.

**Review by ggainey — APPROVED** [review]

**Pull request merged**